quickwit/quickwit-indexing/src/models/pipeline_id.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+/// Uniquely identifies an indexing pipeline.
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct IndexingPipelineId {
+    pub index_id: String,
+    pub source_id: String,
+    pub node_id: String,
+    pub pipeline_ord: usize,
+}
+
+impl IndexingPipelineId {
+    pub fn new(index_id: &str, source_id: &str, node_id: &str, pipeline_ord: usize) -> Self {
```

Comment:
Though not a big deal here, in general, `fn new(index_id: String, ...)` is better for a constructor because it gives the caller the opportunity to pass an owned string that he may no longer need.

We can add a  `#[cfg(test)] fn for_test(index_id: &str, ...)` for convenience.